packages/backend/src/apps/vault-workspace/common/filter-table-rows.ts

```diff
@@ -38,7 +38,11 @@ const filterTableRows = async (
     if (name === VAULT_ID) {
       row[name] = rawData[name]
     } else {
-      row[mapping[name]] = rawData[name]
+      // keys are converted to hex here to satisfy our requirement for template
+      // keys to be alphanumeric, and will be decoded by getDataOutMetadata before
+      // displayed on frontend
+      const key = Buffer.from(mapping[name]).toString('hex')
```

Comment:
This a bit tricky because this will break for existing published pipes

Example:
* Before this PR the pipe uses this vault variable: `{{step.uuid-000-abc.some_column}}`, and the vault action's `dataOut` looks like 
```
{
  some_column: 'col_value'
}
```
* After this PR, the vault action's `dataOut` will become like this in future pipe executions:
```
{
  736f6d655f636f6c756d6e: 'col_value'
}
```
This means we actually need to migrate old variables: `{{step.uuid-000-abc.<column_name>}}` to `{{step.uuid-000-abc.hexEncode(<column_name>)}}` which is overcomplex.

I'm thinking:
- We know that our variable regex currently only accepts `[\da-zA-Z-_]+`, so we should hexEncode / hexDecode only if the column name doesn't pass this regex.
- Or if you think this is overcomplex, we can just modify the variable regexp as well. Feels like this might also be a dangerous change. Can jsut choose the one least likely to blow up.